pages/index.tsx

```diff
@@ -110,7 +115,47 @@ const HomePage = (props: any) => {
     props.editMode
   )
 
+  const [authPopupDisplayed, setAuthPopupDisplayed] = useState(false)
+
+  const refreshPage = () => {
+    fetch(`/api/reset-preview`).then(() => {
+      window.location.href = "/?autoAuth"
+    })
+  }
+
+  useEffect(() => {
+    if (window.location.href.includes("autoAuth")) {
+      setAuthPopupDisplayed(true)
+    }
+  })
+
+  
+  
+
   return (
+    <InlineForm form={form}>
+      {authPopupDisplayed && (
+        <ActionableModal title="Authentication" message="To edit this site, you first need to be authenticated." actions={[
+          {
+            name: "Continue",
+            action: enterEditMode,
+          },
+          {
+            name: "Cancel",
+            action: null,
```

Comment:
It's a bit weird that we do a check for autoAuthPopupShow here, and also in side the modal itself. Maybe instead, the action passed could be 
```
() => {
setAuthPopupDisplayed(false)
}
```
Then the modal is no longer responsible for a separate open/closed state